src/test/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationTest.java

```diff
@@ -241,4 +241,44 @@ private Person createTestPerson(String matriculationNumber, String name) {
                 new HashSet<>()
         );
     }
+
+    @Test
+    void testSetAndGetAssessmentWeight() {
+        StudentCourseAssociation sca = createTestSCA();
+        sca.setAssessmentWeight("Midterm", 0.4);
+        sca.setAssessmentWeight("Final", 0.6);
+
+        Map<String, Double> weights = sca.getAllWeights();
+        assertEquals(2, weights.size());
+        assertEquals(0.4, weights.get("Midterm"), 0.001);
+        assertEquals(0.6, weights.get("Final"), 0.001);
+    }
+
+    @Test
+    void testGetAllWeights() {
+        StudentCourseAssociation sca = createTestSCA();
+        sca.setAssessmentWeight("Midterm", 0.4);
+        sca.setAssessmentWeight("Final", 0.6);
+
+        Map<String, Double> weights = sca.getAllWeights();
+        assertEquals(2, weights.size());
+        assertTrue(weights.containsKey("Midterm"));
+        assertTrue(weights.containsKey("Final"));
+        assertEquals(0.4, weights.get("Midterm"), 0.001);
+        assertEquals(0.6, weights.get("Final"), 0.001);
+    }
+
+    private StudentCourseAssociation createTestSCA() {
```

Comment:
`SCA` in method names should only have the first letter uppercase, i.e. `createTestSca()`